src/Kernel/client/telemetry.ts

```diff
@@ -34,94 +36,97 @@ class LiteEvent<T> implements ILiteEvent<T> {
-            var script = document.createElement('script');
+        return new Promise((resolve) => {
+            const script = document.createElement('script');
             script.onload = () => { resolve(); };
             script.type = 'text/javascript';
             script.src = javascriptUrl;
             document.head.append(script);
         });
     }
 
-    public async requestConsent(consentMarkup: ConsentMarkup) {
-        console.log(`Adding Cookie API Stylesheets to the document: ${consentMarkup.Stylesheets.length} files`);
-        await Promise.all(consentMarkup.Stylesheets.map((css, _, __) => this.addStylesheet(css)));
-        console.log(`Cookie API Stylesheets loaded.`);
+    private onConsentChanged(categoryPreferences: Record<consentCategories, boolean>) {
+        console.log("onConsentChanged", categoryPreferences);
+        this.checkRequiredConsent();
+    }
+
+    private checkRequiredConsent() {
+        const hasConsent = this.siteConsent.getConsentFor(consentCategories.Required);
+        console.log(`HasConsent: ${hasConsent}`);
+        if (hasConsent) {
+            this.onConsentGranted();
+        }
+    }
 
-        var div = document.createElement('div');
-        div.innerHTML = consentMarkup.Markup;
+    public async requestConsent() {
+        const div = document.createElement('div');
+        div.id = "cookie-banner";
         document.body.prepend(div);
-        console.log("Cookie API markup div added");
+        console.log("Cookie banner div added");
 
-        console.log(`Adding Cookie API Javascripts to the document: ${consentMarkup.Javascripts.length} files`);
-        await Promise.all(consentMarkup.Javascripts.map((javascript, _, __) => this.addJavascript(javascript)));
+        console.log(`Adding Cookie API Javascript to the document`);
+        await this.addJavascript(COOKIE_CONSENT_JS);
         console.log(`Cookie API Javascripts loaded.`);
 
-        this.mscc = mscc;
-        if (this.mscc == null) {
+        if (WcpConsent === null) {
             console.log("Cookie API javascript was not loaded correctly. Consent cannot be obtained.");
             return;
         }
 
-        console.log("Calling Cookie API hasConsent");
-        this.mscc.on('consent', this.onConsentGranted);
-        var hasConsent = this.mscc.hasConsent();
-        console.log(`HasConsent: ${hasConsent}`);
-        if (hasConsent) {
-            this.onConsentGranted();
-        }
+        console.log("Initializing WcpConsent...");
+        const userLanguage = navigator.language;
+        WcpConsent && WcpConsent.init(userLanguage, div.id, (err, siteConsent) => {
+            if (!err) {
+                this.siteConsent = siteConsent;
+                console.log("WcpConsent initialized.");
+                this.checkRequiredConsent();
+            } else {
+                console.log("Error initializing WcpConsent: " + err);
+            }
+        }, this.onConsentChanged, Themes.light);
```

Comment:
This could work poorly if the user has dark mode set in Windows 10 or Android preferences. Using a [CSS media query](https://usefulangle.com/post/318/javascript-check-dark-mode-activated) should allow for detecting and conditioning on whether the dark-mode preference is set.

```suggestion
        }, this.onConsentChanged, window.matchMedia('prefers-color-scheme: dark').matches ? Themes.dark : Themes.light);
```